File: 프로그래머스/Lv.0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.py
def solution(arr, queries):
    answer = []
    for s,e,k in queries:
        a = [arr[num] for num in range(s, e+1) if arr[num] > k]
        if a:
            answer.append(min(a))
        else:
            answer.append(-1)
    
    return answer

File: 프로그래머스/Lv.0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
import java.util.Arrays;

class Solution {
    public int[] solution(int[] arr, int[][] queries) {
        int[] answer = new int[queries.length];
        int cnt = 0;
        for(int[] temp : queries){
            int min = -1;
            for(int i = temp[0]; i <= temp[1]; i++){
                if(temp[2] < arr[i]){
                    min = min == -1 ? arr[i] : (min > arr[i] ? arr[i] : min);
                }
            }
            answer[cnt] = min;
            cnt++;
        }
        return answer;
    }
}